src/web/blocks.rs

```diff
@@ -52,41 +54,49 @@ pub struct BlockDTO {
     height: u32,
     timestamp: Timestamp,
     block_hash: SerScaleHex<Hash>,
-    parent_block_hash: SerScaleHex<Hash>,
-    transactions_merkle_root_hash: SerScaleHex<Hash>,
-    rejected_transactions_merkle_root_hash: SerScaleHex<Hash>,
-    invalidated_blocks_hashes: Vec<SerScaleHex<Hash>>,
-    transactions: Vec<SerScaleHex<VersionedValidTransaction>>,
-    rejected_transactions: Vec<SerScaleHex<VersionedRejectedTransaction>>,
-    view_change_proofs: Vec<SerScaleHex<Hash>>,
+    parent_block_hash: Option<HashOf<VersionedCommittedBlock>>,
+    transactions_merkle_root_hash: Option<HashOf<MerkleTree<VersionedSignedTransaction>>>,
+    rejected_transactions_merkle_root_hash: Option<HashOf<MerkleTree<VersionedSignedTransaction>>>,
+    // invalidated_blocks_hashes: Vec<SerScaleHex<Hash>>,
+    transactions: Vec<TransactionValue>,
+    // rejected_transactions: Vec<SerScaleHex<VersionedSignedTransaction>>,
+    // view_change_proofs: Vec<SerScaleHex<Hash>>,
 }
 
-impl TryFrom<BlockValue> for BlockDTO {
+impl TryFrom<VersionedCommittedBlock> for BlockDTO {
     type Error = color_eyre::Report;
 
-    fn try_from(block: BlockValue) -> Result<Self> {
+    fn try_from(block: VersionedCommittedBlock) -> Result<Self> {
+        let committed_block = block.into_v1();
```

Comment:
```suggestion
        let block = block.into_v1();
```

[Shadowing](https://doc.rust-lang.org/rust-by-example/variable_bindings/scope.html) should work just fine.